documentation/spring/guide-querydsl-spring.asciidoc

```diff
@@ -0,0 +1,45 @@
+:toc: macro
+toc::[]
+
+= QueryDSL in devon4j-spring
+
+To implement dynamic queries, devon4j suggests the use of QueryDSL. QueryDSL uses metaclasses generated from entity classes at build time.  devon4j-spring provides a way to use QueryDSL without the need for code generation. For this, devon4j provides the interface https://github.com/devonfw/devon4j/blob/master/modules/jpa-spring-data/src/main/java/com/devonfw/module/jpa/dataaccess/api/data/DefaultRepository.java[DefaultRepository] that your repository needs to extend and the https://github.com/devonfw/devon4j/blob/master/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/QueryUtil.java[QueryUtil] helper class to build your queries.
```

Comment:
this guid is OK, but actually the usage of QueryDSL without code-generation is not devon4j specific. However, it will not work with quarkus native-image (yet). Anyhow fine for now.